test/estimate-gas.test.js

```diff
@@ -0,0 +1,138 @@
+            await ctx.contracts.atonomi.registerAndActivateDevice(deviceId, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            await ctx.contracts.atonomi.updateReputationScore(deviceId, score, {from: ctx.actors.irnNode})
+            const testWithdraw = { account: ctx.actors.mfg, expectedTokens: mfgReward }
+            const from = testWithdraw.account
+            var gas  = await ctx.contracts.atonomi.withdrawTokens.estimateGas({ from: from })
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'mfgr', task: 'withdraw tokens', gas: gas } )
+        })
+    })
+
+    describe('owner costs', () => {
+        it('   ^= to Activate Device', async() => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, regFee, { from: ctx.actors.mfg })
+            await ctx.contracts.atonomi.registerDevice(deviceIdHash, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, actFee, { from: ctx.actors.deviceOwner })
+            var gas = await ctx.contracts.atonomi.activateDevice.estimateGas(deviceId, {from: ctx.actors.deviceOwner})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'owner', task: 'activate device', gas: gas } )
+        })
+    })
+    
+    describe('reputation author costs', () => {
+        it('   ^= to Submit Reputation', async () => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, regFee + actFee, {from: ctx.actors.mfg})
+            await ctx.contracts.atonomi.registerAndActivateDevice(deviceId, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            const gas = await ctx.contracts.atonomi.updateReputationScore.estimateGas(deviceId, score, {from: ctx.actors.irnNode})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'reputation author', task: 'submit reputation', gas: gas } )
+        })
+        it('   ^= to Withdraw Tokens', async() => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, regFee + actFee, {from: ctx.actors.mfg})
+            await ctx.contracts.atonomi.registerAndActivateDevice(deviceId, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            await ctx.contracts.atonomi.updateReputationScore(deviceId, score, {from: ctx.actors.irnNode})
+            const testWithdraw = { account: ctx.actors.irnNode, expectedTokens: irnReward }
+            var gas  = await ctx.contracts.atonomi.withdrawTokens.estimateGas({ from: testWithdraw.account })
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'reputation author', task: 'withdraw tokens', gas: gas } )
+        })
+    })
+
+    describe('writing output', () => {
+        it('to CSV...', () => {
+            csvWriter.writeRecords(recordsForCSV)
```

Comment:
If this describe block always runs last, then shouldnt be an issue, otherwise may be better to append records as tests exectue